linters/errcheck: add support for ignore/exclude flags

a8m opened the pull request:
Follow golangci/golangci-lint#223, I've added support for the `exclude` and `ignore` flags of errcheck.
Note that, this commit depends on golangci/errcheck#1. We need to merge it first, update vendors and only then merge this one.

Update: all lint errors will disappear after merging golangci/errcheck#1.

a8m commented:
Any update on this @jirfag?

jirfag commented:
hi!
thank you for the pull request and sorry for the delay. I've merged errcheck, update it in for the `dep` please.

jirfag commented on pkg/commands/run.go:
```diff
@@ -75,16 +75,20 @@ func initFlagSet(fs *pflag.FlagSet, cfg *config.Config, m *lintersdb.Manager) {
 	hideFlag("errcheck.check-blank")
+	fs.StringVar(&lsc.Errcheck.Exclude, "errcheck.exclude", "", "errcheck.exclude")
+	hideFlag("errcheck.exclude")
+	lsc.Errcheck.Ignore = config.IgnoreFlag{}
```
set it in `defaultLintersSettings` please

jirfag commented on pkg/commands/run.go:
```diff
@@ -75,16 +75,20 @@ func initFlagSet(fs *pflag.FlagSet, cfg *config.Config, m *lintersdb.Manager) {
 	fs.BoolVar(&lsc.Errcheck.CheckAssignToBlank, "errcheck.check-blank", false,
 		"Errcheck: check for errors assigned to blank identifier: _ = errFunc()")
 	hideFlag("errcheck.check-blank")
+	fs.StringVar(&lsc.Errcheck.Exclude, "errcheck.exclude", "", "errcheck.exclude")
```
flags creation isn't needed: flags above are preserved just for the compatibility with users who use them.

a8m replied:
IMO, it makes sense to keep the API compatible for both file and CLI configuration.

a8m commented:
Hey @jirfag, thanks for the CR. 
I updated the deps and addressed the CR comments, except the one about the flags (see my comment below). 
Let me know what else do you need in order to land this.

jirfag commented:
thank you!
